cip/CIP2016-07-07-Type-conversion-functions.adoc

```diff
@@ -0,0 +1,79 @@
+= CIP2016-07-07 - Type conversion functions
+:numbered:
+:toc:
+:toc-placement: macro
+:source-highlighter: codemirror
+
+*Author:* Mats Rydberg <mats@neotechnology.com>
+
+[abstract]
+.Abstract
+--
+Cypher contains three type conversion functions `toString()`, `toInt()`, and `toFloat()`.
+This CIP suggests introducing a more comprehensive rule for specifying the existence of these functions, following a simple `toX()` pattern, where `X` is the name of the type.
+Further, this CIP suggests expanding this rule to encompass all scalar types.
+--
+
+toc::[]
+
+
+== Motivation
+
+The `toInt()` function indicates conversion to something called `Int`, which does not exist in Cypher.
+Presumably every programmer will be aware that `int` is a common short form for `integer`, but this relies on an inherited, implicit piece of knowledge.
+Following Cypher's guiding principles, explicit is better than implicit, and this CIP argues that `toInteger()` is a more explicit and better name.
+
+Furthermore, there is an opportunity to create a very simple and consistent rule that all scalar types should come with a type conversion function on the pattern `toX()` where `X` is the name of the type.
+This has the consequence of adding a `toBoolean()` function to the language, creating a full set of conversion functions.
+
+== Proposal
+
+In concrete terms, this CIP suggests two things:
+
+- The function `toInt()` should be renamed to `toInteger()`.
+- The function `toBoolean()` should be added to the language.
+
+=== Syntax
+
+Apart from adding a new function and changing the name of a function, no syntax changes are suggested.
+
+=== Semantics
+
+==== `toInteger()`
+
+The renamed `toInteger()` function would behave exactly the way `toInt()` behaves.
+
+==== `toBoolean()`
+
+The new `toBoolean()` function would only accept the two strings `'true'` and `'false'` as valid inputs, and would return `null` on any other input, similarly to how `toFloat()` and `toInt()` behave at the time of writing.
+
+=== Examples
+
+.Convert a string to an integer.
+[source, cypher]
+----
+RETURN toInteger('1')
+----
+
+.Convert a float to an integer.
+[source, cypher]
+----
+RETURN toInteger(1.5)
+----
+
+.Convert strings to booleans.
+[source, cypher]
+----
+UNWIND ['true', 'false', 'foo'] AS s
+RETURN toBoolean(s) // yields two booleans and one null
+----
+
+=== Interaction with existing features
+
+With the addition of `toBoolean()`, it is now possible to do a round-trip conversion of booleans to strings (using `toString()`) and back again.
+
+== Benefits to this proposal
```

Comment:
The points below would be far better presented as items in a list. This will make them more readable. If  they are left as they are (which is also an option), they would need to be reworded so that they are proper sentences, which they are not currently.